public/src/app/components/diagnostics/diagnostics.component.html

```diff
@@ -0,0 +1,53 @@
+<div class="content-container">
+  <h3 class="upper">Running connectivity diagnostics</h3>
+  <h4>Please wait as we perform some tests</h4>
+  <form [formGroup]="testsForm">
+    <input type="checkbox" formControlName="{{Protocol.TCP}}" /><label for="cb"></label>
+    <span
+      >Connection to <strong>amq.nexploit.app</strong> at
+      <strong>port 5672</strong></span
+    ><br />
+    <input type="checkbox" formControlName="{{Protocol.HTTP}}" /><label for="cb"></label>
+    <span
+      >Connection to <strong>nexploit.app</strong> at
+      <strong>port 443</strong></span
+    >
+    <button class="button-scan button-move" (click)="restartTest()">
+      Restart Test</button
+    ><br />
+    <input type="checkbox" formControlName="{{Protocol.AUTH}}" /><label for="cb"></label>
+    <span>Valid authentication credentials</span>
+  </form>
+  <h4 class="upper">Progress</h4>
+  <ng-container>
+    <div class="form-control">
+      <div>
+        <ng-template ngFor let-type [ngForOf]="protocolTypes" let-i="index">
+          <span class="span-msg">{{ connectivityMessages[i] }}</span
+          ><br />
+          <ng-template [ngIf]="responseMessage">
+            <span
+              id="type"
+              class="span-msg"
+              [ngClass]="
+                responseMessage[type] === 'Success' ? 'success' : 'fail'
+              "
+              >{{ responseMessage[type] }}</span
```

Comment:
Instead of `this.responseMessage[type] = this.protocolMessage.transform(type);` you should use pipe in templates
```suggestion
              >{{ type | protocol }}</span
```